File: codeforce/yandex_algo_2017_r1/pB.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    long long m, n;
    cin >> m >> n;
    long long ans = 0, a, b;
    for (long long i = 0 ; i < 2000 ; i++)
        if ((m - n - i) >= n){
            a = (5000 - ((m - n - i) % 5000)) % 500;
            b = (5000 - ((n+i) % 5000)) % 500;
            ans = max(a+b, ans);
        }
    cout << ans << endl;
}

File: codeforce/548_div2/pB.py
n = int(input())
nums = list(map(int, input().split()))

cur = 1e18
ans = 0
for idx in range(n-1, -1, -1):
    cur = min(cur-1, nums[idx])
    cur = max(0, cur)
    ans += cur
print(ans)
